lib/wasi/src/runtime/module_cache/mod.rs

```diff
@@ -0,0 +1,38 @@
+mod and_then;
+mod disabled;
+mod on_disk;
+mod shared;
+mod thread_local;
+mod types;
+
+pub use self::{
+    and_then::AndThen,
+    on_disk::OnDiskCache,
+    shared::SharedCache,
+    thread_local::ThreadLocalCache,
+    types::{CacheError, Key, ModuleCache},
+};
+
+pub(crate) use self::disabled::Disabled;
+
+/// Get a [`ModuleCache`] which should be good enough for most in-memory use
+/// cases.
+///
+/// # Platform-specific Notes
+///
+/// This will use the [`ThreadLocalCache`] when running in the browser.  Each
+/// thread lives in a separate worker, so sharing compiled modules in the
+/// browser requires using a custom [`ModuleCache`] built on top of
+/// [`postMessage()`][pm] and [`SharedArrayBuffer`][sab].
+///
+/// [pm]: https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage
+/// [sab]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer
+pub fn in_memory() -> impl ModuleCache + Send + Sync {
```

Comment:
I don't think we should do this. 

It would require either adding a `cache_dir: PathBuf` argument that is completely useless when running on the web, or we would need to duplicate `wasmer-registry`'s logic for determining the cache directory.

Whatever mechanism people use to discover the `in_memory()` method (auto-complete, API docs, etc.) will also show them there is a `FileSystemCache`, so I don't think discoverability will be a big deal.